File: evidence/generated-systems/angular/governance-on-angular/src/app/models/AuditProgram.js

// Define collection and schema for AuditProgram
export interface AuditProgram {
    name:
	type : string
    scope:
	type : string
    Organization:
	type : Schema.Types.ObjectId
    Engagements:
 	type : [{ type: Schema.Types.ObjectId, ref: 'AuditEngagement' }]
    Cycle:
 	type : String
    Status:
 	type : String
#
    collection: 'auditPrograms'
}

File: evidence/generated-systems/rubyonrails/governance-on-rails/governanceonrails/app/models/auditprogram.rb
class AuditProgram < ApplicationRecord
  enum Cycle: [:Annual, :SemiAnnual, :Quarterly, :Continuous, :OneTime]
  enum Status: [:Planned, :Fieldwork, :Reporting, :Closed, :OnHold]


  has_many :Organization, class_name: 'Organization'
  has_many :Engagements, class_name: 'AuditEngagement'

end
